apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/dao/state/poll/PollTable.java

```diff
@@ -0,0 +1,211 @@
+            pstmt.setLong(++i, poll.getId());
+            pstmt.setLong(++i, poll.getAccountId());
+            pstmt.setString(++i, poll.getName());
+            pstmt.setString(++i, poll.getDescription());
+            DbUtils.setArray(pstmt, ++i, poll.getOptions());
+            pstmt.setInt(++i, poll.getFinishHeight());
+            pstmt.setByte(++i, poll.getVoteWeighting().getVotingModel().getCode());
+            DbUtils.setLongZeroToNull(pstmt, ++i, poll.getVoteWeighting().getMinBalance());
+            pstmt.setByte(++i, poll.getVoteWeighting().getMinBalanceModel().getCode());
+            DbUtils.setLongZeroToNull(pstmt, ++i, poll.getVoteWeighting().getHoldingId());
+            pstmt.setByte(++i, poll.getMinNumberOfOptions());
+            pstmt.setByte(++i, poll.getMaxNumberOfOptions());
+            pstmt.setByte(++i, poll.getMinRangeValue());
+            pstmt.setByte(++i, poll.getMaxRangeValue());
+            pstmt.setInt(++i, poll.getTimestamp());
+            pstmt.setInt(++i, poll.getHeight());
+            pstmt.executeUpdate();
+        }
+    }
+
+    public Poll getPoll(long id) {
+        return get(POLL_LONG_KEY_FACTORY.newKey(id));
+    }
+
+    public DbIterator<Poll> getPollsFinishingAtOrBefore(int height, int from, int to) {
+        return getManyBy(new DbClause.IntClause(FINISH_HEIGHT, DbClause.Op.LTE, height), from, to);
+    }
+
+    public DbIterator<Poll> getPollsFinishingBelowHeight(int height, int from, int to) {
+        // select all Polls where 'finish_height' is LESS (DbClause.Op.LT) then specified height value
+        return getManyBy(new DbClause.IntClause("finish_height", DbClause.Op.LT, height), from, to);
+    }
+
+    public DbIterator<Poll> getPollsFinishingAtHeight(int height) {
+        // EXACT matching to Poll finish height
+        return getManyBy(new DbClause.IntClause("finish_height", height), 0, Integer.MAX_VALUE);
+    }
+
+    public DbIterator<Poll> getActivePolls(int from, int to, int height) {
+        return getManyBy(new DbClause.IntClause("finish_height", DbClause.Op.GT, height), from, to);
+    }
+
+    public DbIterator<Poll> getPollsByAccount(
+        long accountId,
+        boolean includeFinished,
+        boolean finishedOnly,
+        int from,
+        int to,
+        int height
+    ) {
+        DbClause dbClause = new DbClause.LongClause("account_id", accountId);
+        if (finishedOnly) {
+            dbClause = dbClause.and(new DbClause.IntClause("finish_height", DbClause.Op.LTE, height));
+        } else if (!includeFinished) {
+            dbClause = dbClause.and(new DbClause.IntClause("finish_height", DbClause.Op.GT, height));
+        }
+        return getManyBy(dbClause, from, to);
+    }
+
+    public DbIterator<Poll> getVotedPollsByAccount(
+        final DataSource dataSource,
+        final long accountId,
+        final int from,
+        final int to
+    ) throws AplException.NotValidException {
+        Connection connection = null;
```

Comment:
Let's avoid exposing 'dataSource' for higher levels and remove parameter in method declaration.
You can take main datasource inside Table component like:
```
TransactionalDataSource dataSource = databaseManager.getDataSource();
connection = dataSource.getConnection();
```

Reply:
done